bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -127,15 +127,14 @@ static inline u64 calc_delta(u64 *prev_val, u64 *val)
 static inline u64 get_on_cpu_cycles(u32 *cpu_id)
 {
 	u64 delta, val, *prev_val;
-	struct bpf_perf_event_value c;
 	int error;
 
 	// TODO: Fix Verifier errors upon changing this to bpf_perf_event_read_value
 	error = bpf_perf_event_read(&cpu_cycles_event_reader, *cpu_id);
 	if (error < 0) {
 		return delta;
 	}
-	val = c.counter;
+	val = error;
```

Comment:
could you please help me understand how this fixes the issue?
I see that `get_on_cpu_instr` also makes use of `c.counter`. Should that be changed as well?

Reply:
looking at the [old code](https://github.com/sustainable-computing-io/kepler/blob/4e8abc8d09cc54b57bc26d95a12c5e938862bda2/bpfassets/libbpf/src/kepler.bpf.c#L116-L141), function `bpf_perf_event_read_value` is getting used in the newer kernels, which i assume to be the case for openshift node, and is also the recommended function over `bpf_perf_event_read`

Reply:
That's not entirely true. `BPF_PERF_EVENT_READ_VALUE_AVAILABLE` was always true given it's a compile time check + all CI systems (and likely all developer systems) are using > 5.12 kernels.

`bpf_perf_event_read_value` is the desired helper to use here, but see #1411 as to why that's not being used here.

This bug stems from me attempting to use ☝️ but not backing that change out completely.
The other comments are reasonable, I've forced pushed a commit that implements those.